app/services/openid_connect_service.rb

```diff
@@ -0,0 +1,12 @@
+class OpenidConnectService
+  def self.generate_authorization_code(user:, client_id:, nonce:)
+    identity = user.identities.create!(
+      service_provider: client_id,
+      session_uuid: SecureRandom.uuid,
+      last_authenticated_at: Time.current,
+      nonce: nonce
+    )
+
+    identity.session_uuid
+  end
```

Comment:
The method is called `generate_authorization_code` but I can't easily tell where a code is generated and returned. This class is creating an identity and returning its `session_uuid`.